ui/components/Filters.js

```diff
@@ -336,44 +338,31 @@ function MesheryFilters({
    * publish filter capability and setting the canPublishFilter state accordingly
    */
   useEffect(() => {
-    dataFetch(
-      '/api/schema/resource/filter',
-      {
-        method: 'GET',
-        credentials: 'include',
-      },
-      (result) => {
-        setImportSchema(result);
-      },
-      handleError(ACTION_TYPES.SCHEMA_FETCH),
-    );
-    dataFetch(
-      '/api/schema/resource/publish',
-      {
-        method: 'GET',
-        credentials: 'include',
-      },
-      async (result) => {
-        try {
-          const { models } = await getMeshModels();
-          const modelNames = _.uniq(models?.map((model) => model.displayName));
-          modelNames.sort();
-
-          // Modify the schema using the utility function
-          const modifiedSchema = modifyRJSFSchema(
-            result.rjsfSchema,
-            'properties.compatibility.items.enum',
-            modelNames,
-          );
-          setPublishSchema({ rjsfSchema: modifiedSchema, uiSchema: result.uiSchema });
-          setMeshModels(models);
-        } catch (err) {
-          console.error(err);
-          setPublishSchema(result);
-        }
-      },
-      handleError(ACTION_TYPES.SCHEMA_FETCH),
-    );
+    async function fetchMeshModels() {
+      try {
+        const { models } = await getMeshModels();
+        const modelNames = models?.map((model) => model.displayName) || [];
+        modelNames.sort(); // Sort model names
+
+        // Modify the schema to include mesh models
+        const modifiedSchema = modifyRJSFSchema(
+          publishCatalogItemSchema, // Use publishSchema as the base schema
+          'properties.compatibility.items.enum',
+          modelNames,
+        );
+
+        // Set the modified schema and UI schema
+        setPublishSchema({ rjsfSchema: modifiedSchema, uiSchema: publishCatalogItemUiSchema });
```

Comment:
@11happy please see here in this state you are having object contains two properties rjsfSchema and uiSchema which is good however checkout this line, https://github.com/meshery/meshery/pull/10720/files#diff-6ec8e150ffa5db5b0830cd1598d529bbc98c79c98df32092858a51a779fcdea0R1471 over here you are just passing state itself instead you should pass it like 

```
      schema={publishFormSchema.rjsfSchema}
      uiSchema={publishFormSchema.uiSchema}
```